### PR TITLE
[Merged by Bors] - refactor(Algebra/Polynomial/Laurent): use Int.ofNat.inj

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
Not sure if we want to have a special name for a theorem of type
`Function.Injective Int.ofNat`
which is just `Int.ofNat.inj` with explicit arguments. If yes, what would be a good name and place for it?

File: Mathlib/Algebra/Polynomial/Laurent.lean [hunk 1]
```diff
@@ -351,12 +351,10 @@ def trunc : R[T;T⁻¹] →+ R[X] :=
 
 @[simp]
 theorem trunc_C_mul_T (n : ℤ) (r : R) : trunc (C r * T n) = ite (0 ≤ n) (monomial n.toNat r) 0 := by
-  -- Porting note: added. Should move elsewhere after the port.
-  have : Function.Injective Int.ofNat := fun x y h => Int.ofNat_inj.mp h
   apply (toFinsuppIso R).injective
   rw [← single_eq_C_mul_T, trunc, AddMonoidHom.coe_comp, Function.comp_apply]
   -- Porting note (#10691): was `rw`
-  erw [comapDomain.addMonoidHom_apply this]
+  erw [comapDomain.addMonoidHom_apply @Int.ofNat.inj]
```
Comment (grunweg) [inline Mathlib/Algebra/Polynomial/Laurent.lean:357]: Do you need the @?

**Review by grunweg — APPROVED** [review]
Looks good to me, pending one question (see below). Thanks!

File: Mathlib/Algebra/Polynomial/Laurent.lean (hunk 1 above)
Comment (mo271) [reply Mathlib/Algebra/Polynomial/Laurent.lean:357]: yes, without it I get
```
application type mismatch
  comapDomain.addMonoidHom_apply Int.ofNat.inj
argument
  Int.ofNat.inj
has type
  Int.ofNat ?m.79663 = Int.ofNat ?m.79664 → ?m.79663 = ?m.79664 : Prop
but is expected to have type
  Injective ?m.79662 : Prop
  ```

**Comment by mo271** [comment]
Perhaps this is a use case for using `alias`?

**Comment by Ruben-VandeVelde** [comment]
Actually, I do think we should have the Injective lemma as well

**Comment by mo271** [comment]
> Actually, I do think we should have the Injective lemma as well

What would be a good name and place for it?

**Comment by Ruben-VandeVelde** [comment]
`Int.ofNat_injective`, and near `Int.natCast_inj` in `Data.Int.Defs`, perhaps

**Comment by mo271** [comment]
> `Int.ofNat_injective`, and near `Int.natCast_inj` in `Data.Int.Defs`, perhaps

done

**Comment by urkud** [comment]
I'm going to test if `Nat.cast_injective` works better in these `rw`s.

**Comment by urkud** [comment]
Thanks! Once CI is happy, please merge this PR using `bors r+`
bors d+

**Pull request closed without merging**